### ISSUE TITLE
After permissions are set to Denу, PermissionRequested event is not raised any longer

### ISSUE THREAD

**Issue by ynborokh** [MicrosoftEdge/WebView2Feedback#2672]
**Description**
Once a permission state is set to `Microsoft.Web.WebView2.Core.CoreWebView2PermissionState.Deny`, `CoreWebView2.PermissionRequested` event is not raised for this permission.

**Version**
SDK: Win App SDK 1.1.3
Framework: WinUI 3
OS: Windows 10 Pro 21H2 19044.1826

**Repro Steps**
1. Use WebView2 in WinUI 3 project. Set `webView2.Source` to a link generated by `https://meet.google.com/` (it requires camera and microphone permissions).
2. Register a handler for `PermissionsRequested` event:
```
webView2.CoreWebView2.PermissionRequested += CoreWebView2_PermissionRequested.
...
private void CoreWebView2_PermissionRequested(
        Microsoft.Web.WebView2.Core.CoreWebView2 sender,
        Microsoft.Web.WebView2.Core.CoreWebView2PermissionRequestedEventArgs args)
    {
        args.State = Microsoft.Web.WebView2.Core.CoreWebView2PermissionState.Deny;
    }
```
3. Once the permission is denied, this choice is remembered and `CoreWebView2_PermissionRequested` will no longer called if we relaunch the application.
4. It seems like a bug, since if I set permission to `Allow` or `Default`, the event is still raised each time I visit the page.

**Comment by IvanKarpan** [comment]
This is frustrating... Especially since there is no access to the default profile object. If there was - it potentially could be possible to reset permissions data for a domain at least.

**Comment by nishitha-burman** [comment]
Hi @ynborokh and @IvanKarpan,

Can you describe why you would like the event to be fired even if it is set to denied?

Thank you,
NIshitha

**Comment by IvanKarpan** [comment]
Hello @nishitha-burman!

When working with a browser the user has the option to manage permissions, so the initial denial of a permission is reversible.

WinUI 3 doesn't expose the profile object the WebView2 is working with... So in case the end-user denied a permission they basically have to completely remove the application's data to get another chance and allow it.

So I see two potential solutions to the problem:
1. either send the `CoreWebView2.PermissionRequested` event every time
2. or allow access and management of the profile object, AFAIU this is where permissions can be managed directly, at least to allow reverting a permission to the default state which in turn will trigger the event

**Comment by PrakashC1** [comment]
Hello @nishitha-burman ad team,
     I am also having the same problem. Could you make this bug as priority and provide some solution, I am expecting it in coming releases..

**Comment by david-risney** [comment]
There's an upcoming API https://github.com/MicrosoftEdge/WebView2Feedback/pull/3065 that should help with this issue

**Comment by vickiez** [comment]
Hi @ynborokh, @IvanKarpan, and @PrakashC1,

We added an experimental [SavesInProfile](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2permissionrequestedeventargs.savesinprofile?view=webview2-dotnet-1.0.1619-prerelease#microsoft-web-webview2-core-corewebview2permissionrequestedeventargs-savesinprofile) property to PermissionRequestedEventArgs in the latest prelease package, 1.0.1619-prerelease. You can use this to control whether permission state is saved to the profile. We also added experimental [SetPermissionStateAsync](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2profile.setpermissionstateasync?view=webview2-dotnet-1.0.1619-prerelease) and [GetNonDefaultPermissionSettingsAsync](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2profile.getnondefaultpermissionsettingsasync?view=webview2-dotnet-1.0.1619-prerelease#microsoft-web-webview2-core-corewebview2profile-getnondefaultpermissionsettingsasync) APIs that you may find useful.

Steps for testing upcoming APIs/features:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/set-preview-channel

Permission management spec: https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/PermissionManagement.md

**Comment by vickiez** [comment]
Hi all, the extended permission APIs are now available in the latest release package, 1.0.1661.34. Closing this issue as completed but feel free to reactivate if needed. Thanks!
